docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,286 @@
+
+Enhancement suggestions are tracked as [GitHub
+issues](https://guides.github.com/features/issues/).  When you are
+creating an enhancement suggestion, please **include as many details
+as possible** as you fill in the template.
+
+* **Use a clear and descriptive title** for the issue to identify the
+  suggestion.
+* **Provide a step-by-step description of the suggested enhancement**
+  in as many details as possible.
+* **Explain why this enhancement would be useful** to most SpECTRE users.
+
+### Your First Code Contribution
+
+Unsure where to begin contributing to SpECTRE? You can start by
+looking through these `good first issue` and `help wanted` issues:
+
+* [Good first issues][good-first-issue] - issues which should only
+  require a few lines of code, and a test or two.
+* [Help wanted issues][help-wanted] - issues which should be a bit
+  more involved than `good first issue`s.
+
+#### Local development
+
+SpECTRE can be developed locally. For instructions on how to do this,
+see the following sections in the [SpECTRE
+documentation](https://spectre-code.org/):
+
+* [Installing SpECTRE](installation.html)
+* [Running Status Checks Locally](travis_guide.html)
+
+### Pull Requests
+
+The process described here has several goals:
+
+- Maintain SpECTRE's quality
+- Fix problems that are important to users
+- Engage the community in working toward the best possible code
+- Enable a sustainable system for SpECTRE's maintainers to review
+  contributions
+
+Please follow these steps to have your contribution considered by the
+maintainers:
+
+1. Follow all instructions in the template.
+2. Follow the \ref code_review_guide "code review guidelines", the
+  \ref writing_unit_tests "guide to writing unit tests", and the \ref
+  writing_good_dox "guide to writing documentation"
+3. After you submit your pull request, verify that all [status
+   checks](https://help.github.com/articles/about-status-checks/) are
+   passing
+
+   > If a status check is failing, and you believe that the failure is
+   > unrelated to your change, please leave a comment on the pull
+   > request explaining why you believe the failure is unrelated. A
+   > maintainer will re-run the status check for you. If we conclude
+   > that the failure was a false positive, then we will open an issue
+   > to track that problem with our status check suite.
+
+While the prerequisites above must be satisfied prior to having your
+pull request reviewed, the reviewers may ask you to complete
+additional design work, tests, or other changes before your pull
+request can be ultimately accepted.
+
+#### How SpECTRE pull request reviews are conducted
+
+> Note that these are guidelines and not rigid rules.
+
+In general, longer pull requests take longer to review, with the time
+scaling exponentially with the number of lines changed.  Therefore if
+your pull request is too large, we may ask you to break it up into
+several smaller pull requests.
```

Comment:
I would also suggest preemptively making pull requests as small as reasonable, as these are the easiest to review.